implementations/rust/ockam/ockam_entity/src/channel/initiator.rs

```diff
@@ -0,0 +1,184 @@
+            msg.proof(),
+        )?;
+
+        if !verified {
+            return Err(EntityError::SecureChannelVerificationFailed.into());
+        }
+        info!(
+            "Verified SecureChannel from: {}",
+            contact.identifier().to_string_representation()
+        );
+
+        let contact = profile.to_contact();
+        let proof = profile.generate_authentication_proof(&channel.auth_hash())?;
+
+        let channel_local_address: Address = random();
+        let channel_remote_address: Address = random();
+        let initiator = Self {
+            local_secure_channel_address: channel.address(),
+            remote_profile_secure_channel_address: None,
+            self_local_address: channel_local_address.clone(),
+            self_remote_address: channel_remote_address.clone(),
+            callback_address: child_ctx.address(),
+        };
+        debug!(
+            "Starting ProfileSecureChannel Initiator at local: {}, remote: {}",
+            &channel_local_address, &channel_remote_address
+        );
+        ctx.start_worker(
+            vec![
+                channel_local_address.clone(),
+                channel_remote_address.clone(),
+            ],
+            initiator,
+        )
+        .await?;
+
+        let auth_msg = ChannelAuthResponse::new(contact, proof, channel_remote_address);
+        child_ctx.send(return_route, auth_msg).await?;
+        debug!("Sent Authentication response");
+
+        let _ = child_ctx.receive::<Confirm>().await?;
+
+        Ok(channel_local_address)
+    }
+}
+
+#[async_trait]
+impl Worker for Initiator {
+    type Message = Any;
+    type Context = Context;
+
+    async fn handle_message(
+        &mut self,
+        ctx: &mut Self::Context,
+        msg: Routed<Self::Message>,
+    ) -> Result<()> {
+        let msg_addr = msg.msg_addr();
+        let msg = msg.into_transport_message();
+        let payload = msg.payload;
+        let mut onward_route = msg.onward_route;
+        let mut return_route = msg.return_route;
+
+        if msg_addr == self.self_local_address {
+            debug!("ProfileSecureChannel Initiator received Encrypt");
+            let remote_profile_secure_channel_address =
+                self.remote_profile_secure_channel_address.clone().unwrap(); // FIXME
+
+            // Send to the other party
+            let _ = onward_route.step()?;
+            let onward_route = onward_route
+                .modify()
+                .prepend(remote_profile_secure_channel_address)
+                .prepend(self.local_secure_channel_address.clone())
+                .into();
+
+            let return_route = return_route
+                .modify()
+                .append(self.self_remote_address.clone())
+                .into();
+
+            let transport_msg = TransportMessage {
+                version: 1,
+                onward_route,
+                return_route,
+                payload,
+            };
```

Comment:
A `TransportMessage` new or create is probably appropriate now. This construct seems to be common